### PR TITLE
[WIP] Introduce postgresql cdc

### PR DESCRIPTION
**Pull request by s7monk**
close https://github.com/apache/incubator-paimon/issues/2300

### Purpose
introduce postgresql cdc

### CODE REVIEW

**Comment by JingsongLi** [comment]
Thanks @s7monk , please add documentation too.

File: paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/postgresql/PostgreSqlDebeziumJsonEventParser.java [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+import java.util.Base64;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.apache.paimon.utils.Preconditions.checkArgument;
+
+/** {@link EventParser} for PostgreSQL Debezium JSON. */
+public class PostgreSqlDebeziumJsonEventParser implements EventParser<String> {
```
Comment (zhuangchong) [inline paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/postgresql/PostgreSqlDebeziumJsonEventParser.java:56]: Can you refer to the MySqlRecordParser class for updates?

File: paimon-flink/paimon-flink-cdc/src/test/java/org/apache/paimon/flink/action/cdc/postgresql/PostgreSqlSyncDatabaseActionITCase.java [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+import java.sql.Statement;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+/** IT cases for {@link PostgreSqlSyncDatabaseAction}. */
+public class PostgreSqlSyncDatabaseActionITCase extends PostgreSqlActionITCaseBase {
```
Comment (zhuangchong) [inline paimon-flink/paimon-flink-cdc/src/test/java/org/apache/paimon/flink/action/cdc/postgresql/PostgreSqlSyncDatabaseActionITCase.java:40]: Need more test cases.

Comment (s7monk) [reply paimon-flink/paimon-flink-cdc/src/test/java/org/apache/paimon/flink/action/cdc/postgresql/PostgreSqlSyncDatabaseActionITCase.java:40]: Not finished yet.

**Comment by s7monk** [comment]
> Thanks @s7monk , please add documentation too.

ok.

File: paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/postgresql/PostgreSqlDebeziumJsonEventParser.java (hunk 1 above)
Comment (s7monk) [reply paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/postgresql/PostgreSqlDebeziumJsonEventParser.java:56]: ok.

**Comment by JingsongLi** [comment]
Looks like finish in https://github.com/apache/incubator-paimon/pull/2734

**Pull request closed without merging**